src/store/clipboard/index.ts

```diff
@@ -3,28 +3,58 @@ import {
 } from 'vuex'
 import { IRootState } from '@/store'
 
+export interface BlocksData {
+  isFocused: boolean;
+  prompt: object;
+}
+
 export interface IClipboardState {
-  simulatorActive: boolean;
+  isSimulatorActive: boolean;
+  blocksData: BlocksData[];
 }
 
 export const stateFactory = (): IClipboardState => ({
-  simulatorActive: false,
+  isSimulatorActive: false,
+  blocksData: [],
 })
 
 export const getters: GetterTree<IClipboardState, IRootState> = {
-  isSimulatorActive: (state) => state.simulatorActive,
+  isSimulatorActive: (state) => state.isSimulatorActive,
+  blocksData: (state) => state.blocksData,
+  getBlockPrompt: (state) => (index) => state.blocksData[index].prompt,
+  isBlockFocused: (state) => (index) => state.blocksData[index].isFocused,
```

Comment:
In the future, I'd prefer to see us avoid these getter-getters (getters that return functions). My personal reasoning: They seem confusing at the call sites because they behave differently than the other state getters (requiring an additional invocation), and they're also misrepresented as computed properties, when in actuality they're not precomputing their values. 

Happy to chat about perspectives I'm missing out on though!

Reply:
In order to get the prompt and focused state of a particular block, I wanted to pass in an index and access it. If it does not make sense to have it as a getter, I can move it to the actions part.